### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by donbonifacio** [puppet-coliseum/mothership#6]
This is a rough roadmap, but I'd like us to define this to create milestones and have a proper view of steps ahead. My suggestion:
- **v0.1.0** : We have the engine with the arena/coordinates/puppets working. The engine will be able to load a level and provide a read interface to it. We'll define a starting set of runes.
- **v0.2.0** : We'll build an UI that shows that level (I don't know if cljs will be ready for reader conditionals by then, but we can also use [play-clj](https://github.com/oakes/play-clj))
- **v0.3.0** : The engine will be able to understand actions. For example, to translate click on coordinate A and on coordinate B, see if valid, if it's a move or an attack, and process it. From my experience, we'll benefit if the bring all possible logic to rules
- **v0.4.0** : Simple AI, that just strikes back

After this, we'll have something to show, we'll hunt for an artist and start defining actual levels.

**Comment by jqmtor** [comment]
I agree with these steps, but I think we should split v0.1.0 in two and leave the UI to the end. The simple string representation will allow to implement all features described in all the three parts and I think it would be good to have a scope as limited as possible. Also it would also allow us to buy some time and increase the possibility of having reader conditionals available.

**Comment by donbonifacio** [comment]
@quimrstorres nice one. I updated the roadmap with that split, and added to v0.1 the definition of a set of starting runes. Maybe we should open an issue to talk about that.
